lib/improver/spotdata/main.py

```diff
@@ -329,37 +306,31 @@ def process_diagnostic(diagnostic, neighbours, sites, forecast_times,
     for a_time in forecast_times:
         # Extract Cube from CubeList at current time.
         time_extract = datetime_constraint(a_time)
-        cube = extract_cube_at_time(cubes, a_time, time_extract)
+        cube = extract_cube_at_time(
+            diagnostic_dict["data"], a_time, time_extract)
         if cube is None:
             # If no cube is available at given time, try the next time.
             continue
 
         ad = {}
-        if additional_diagnostics is not None:
-            # Extract additional diagnostcs at current time.
-            ad = extract_ad_at_time(additional_diagnostics, a_time,
+        if diagnostic_dict["additional_data"] is not None:
+            # Extract additional diagnostics at current time.
+            ad = extract_ad_at_time(diagnostic_dict["additional_data"], a_time,
                                     time_extract)
 
         args = (cube, sites, neighbour_list, ancillary_data, ad)
 
         # Extract diagnostic data using defined method.
         resulting_cubes.append(
             ExtractData(
-                diagnostic['interpolation_method']).process(*args, **kwargs)
-            )
+                diagnostic_dict['interpolation_method']).process(
+                    *args, **kwargs))
 
-    # Concatenate CubeList into Cube, creating a time DimCoord, and write out.
-    if resulting_cubes:
+    if diagnostic_dict['extrema']:
         cube_out, = resulting_cubes.concatenate()
-        WriteOutput('as_netcdf', dir_path=output_path).process(cube_out)
-    else:
-        raise Exception('No data available at given forecast times.')
-
-    # If set in the configuration, extract the diagnostic maxima and minima
-    # values.
-    if diagnostic['extrema']:
         extrema_cubes = ExtractExtrema(24, start_hour=9).process(cube_out)
         extrema_cubes = extrema_cubes.merge()
-        for extrema_cube in extrema_cubes:
-            WriteOutput('as_netcdf', dir_path=output_path).process(
-                extrema_cube)
+    else:
+        extrema_cubes = None
+
+    return resulting_cubes, extrema_cubes
```

Comment:
Return `cube_out` instead of `resulting_cubes` to avoid need to re-concatenate in `improver-spot-extract`.